pkg/dao/rpms.go

```diff
@@ -620,3 +619,73 @@ func (r *rpmDaoImpl) ListSnapshotErrata(ctx context.Context, orgId string, snaps
 
 	return response, total, nil
 }
+
+func (r *rpmDaoImpl) ListTemplateRpms(ctx context.Context, orgId string, templateUUID string, search string, pageOpts api.PaginationData) ([]api.SnapshotRpm, int, error) {
+	response := []api.SnapshotRpm{}
+	pulpHrefs := []string{}
+
+	snapshotUuids, err := FetchSnapshotsForTemplate(ctx, r.db, orgId, templateUUID)
+	if err != nil {
+		return response, 0, err
+	}
+
+	res := readableSnapshots(r.db.WithContext(ctx), orgId).Where("snapshots.UUID in ?", UuidifyStrings(snapshotUuids)).Pluck("version_href", &pulpHrefs)
```

Comment:
Alternatively, FetchSnapshotsForTemplate could return Snapshot objects, and then you could just loop through and pull off the version_href.   That would only result in 1 query (but more data returned for that 1 query)